Sources/MapboxDirectionsCLI/CodingOperation.swift

```diff
@@ -89,21 +131,33 @@ class CodingOperation<ResponceType : Codable, OptionsType : DirectionsOptions >
     
     func execute() throws {
         
-        let input = FileManager.default.contents(atPath: NSString(string: options.inputPath).expandingTildeInPath)!
+//        let input = FileManager.default.contents(atPath: NSString(string: options.inputPath).expandingTildeInPath)!
         let config = FileManager.default.contents(atPath: NSString(string: options.configPath).expandingTildeInPath)!
+        let input: Data!
         
         let decoder = JSONDecoder()
         
         let directionsOptions = try decoder.decode(OptionsType.self, from: config)
         
+        if let inputPath = options.inputPath {
+            input = FileManager.default.contents(atPath: NSString(string: inputPath).expandingTildeInPath)!
+        } else {
+            let response = requestResponse(directionsOptions.waypoints, includesSteps: directionsOptions.includesSteps)
+            input = response.0
+        }
+        
         decoder.userInfo = [.options: directionsOptions,
-                            .credentials: BogusCredentials]
+                            .credentials: credentials]
         
         var routeResponse: RouteResponse?
         if options.outputFormat == .gpx {
-            let gpxData = try String(contentsOfFile: options.inputPath).data(using: .utf8)!
-            routeResponse = try! decoder.decode(RouteResponse.self, from: gpxData)
+            if let gpxData = input {
+                routeResponse = try! decoder.decode(RouteResponse.self, from: gpxData)
+            } else if let gpxData = try String(contentsOfFile: options.inputPath!).data(using: .utf8) {
+                routeResponse = try! decoder.decode(RouteResponse.self, from: gpxData)
```

Comment:
A failure to decode could be something other than an internal error, so instead of force-unwrapping, we should catch the error, print it to stderr, and exit with an error code.